.github/workflows/gen-sui-upgrade-version-pr.yml

```diff
@@ -0,0 +1,111 @@
+name: Generate Sui Version Upgrade PR
+run-name: Generate Sui Version Upgrade PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      new_tag:
+        description: "New Sui tag (e.g., testnet-v1.55.0)"
+        required: true
+        type: string
+
+concurrency: ${{ github.workflow }}-${{ inputs.new_tag }}
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }} # for gh pr create
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # pin@v4
+
+      - name: Update Sui refs (single loop), build, commit, push, PR
+        shell: bash
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -euo pipefail
+
+          NEW_TAG="${{ inputs.new_tag }}"
+          BASE="main"   # <-- always use main as the base branch
+
+          # One list with all relevant TOML locations (globs ok)
+          FILES=(
+            "Move.toml"
+            "crates/**/Move.toml"
+            "contracts/**/Move.toml"
+            "apps/**/Move.toml"
+            "Cargo.toml"
+            "crates/**/Cargo.toml"
+            "apps/**/Cargo.toml"
+            "docker/**/sui_version.toml"
+          )
+
+          STAMP="$(date +%Y%m%d%H%M%S)"
+          BRANCH="chore/bump-sui-${NEW_TAG}-${STAMP}"
+          git checkout -b "$BRANCH"
+
+          shopt -s nullglob globstar
+          changed=0
+
+          # A single sed script with three *independent* substitutions:
+          # 1) Move.toml Sui framework rev (strict: repo + subdir must match)
+          # 2) Cargo.toml Sui git deps tag (strict: same inline table contains the Sui repo URL)
+          # 3) docker/*/sui_version.toml simple key="testnet-vX.Y.Z"
+          for pat in "${FILES[@]}"; do
+            for f in $pat; do
+              [[ -f "$f" ]] || continue
+              before="$(sha1sum "$f" | awk '{print $1}')"
+
+              # Use extended regex, in-place
+              sed -E -i \
```

Comment:
@mlegner note that this script is not portable to BSD (macos). in-place edits in sed are incompatible between BSD and linux.